base/pkg/types.jl

```diff
@@ -40,15 +41,73 @@ show(io::IO, s::VersionSet) = join(io, s.intervals, " ∪ ")
 isempty(s::VersionSet) = all(isempty, s.intervals)
 in(v::VersionNumber, s::VersionSet) = any(i->in(v,i), s.intervals)
 function intersect(A::VersionSet, B::VersionSet)
-    ivals = vec([ intersect(a,b) for a in A.intervals, b in B.intervals ])
-    ivals = copy(ivals) # temporary bandaid for issue #4592
+    ivals = [intersect(a,b) for a in A.intervals for b in B.intervals]
```

Comment:
Yes it's equivalent, this was just removing an outdated band-aid. Sorry, maybe next time I should put such things in a separate commit.